### PR TITLE
Add initial base tower crate

### PR DESCRIPTION
**Pull request by LucioFranco**
This adds the initial base tower crate, as of right now it contains
nothing and is only needed to ensure that cargo workspaces can
properly compile with rust 1.32.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,32 @@
+[package]
+name = "tower"
+version = "0.1.0"
+authors = ["Lucio Franco <luciofranco14@gmail.com>"]
```
Comment (jonhoo) [inline Cargo.toml]: This is super nitpicky, but I'd probably expand this list.
Or set it to the relatively generic `Carl Lerche <me@carllerche.com>` :joy:

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+readme = "README.md"
+repository = "https://github.com/tower-rs/tower"
+description = """
+Tower is a library of modular and reusable components for building robust 
+clients and servers.
+"""
+categories = ["asynchronous", "network-programming"]
+keywords = ["io", "async", "non-blocking", "futures", "service"]
+
+[dependencies]
```
Comment (jonhoo) [inline Cargo.toml]: Why list anything under here at the moment?

File: Cargo.toml [hunk 3]
```diff
@@ -16,11 +42,4 @@ members = [
...
   "tower-watch",
-]
-
-[dev-dependencies]
-futures = "0.1"
-log = "0.4.1"
-env_logger = { version = "0.5.3", default-features = false }
-tokio-timer = "0.1"
-futures-cpupool = "0.1"
+]
```
Comment (jonhoo) [inline Cargo.toml]: Was this intentional?

File: examples/channel_service.rs [hunk 4]
```diff
@@ -77,15 +79,16 @@ impl NewChannelService {
...
 }
 
-impl NewService for NewChannelService {
-    type Request = String;
-    type Response = String;
-    type Error = Error;
-    type InitError = io::Error;
-    type Service = ChannelService;
-    type Future = FutureResult<Self::Service, io::Error>;
+impl Service<()> for NewChannelService {
```
Comment (jonhoo) [inline examples/channel_service.rs]: Err, this seems like an unrelated change?

Comment (jonhoo) [reply examples/channel_service.rs]: Or was it just that it wasn't tested before, and wasn't compiling now that it's tested?

File: Cargo.toml [hunk 5]
```diff
@@ -1,6 +1,32 @@
+[package]
+name = "tower"
+version = "0.1.0"
+authors = ["Carl Lerche <me@carllerche.com>]
```
Comment (jonhoo) [inline Cargo.toml]: Missing `"` at the end here now

Comment (LucioFranco) [reply Cargo.toml]: ah yes! missed that totally thanks!

File: Cargo.toml [hunk 6]
```diff
@@ -1,6 +1,30 @@
...
+authors = ["Carl Lerche <me@carllerche.com>"]
+license = "MIT"
+readme = "README.md"
+repository = "https://github.com/tower-rs/tower"
+description = """
+Tower is a library of modular and reusable components for building robust
+clients and servers.
+"""
+categories = ["asynchronous", "network-programming"]
+keywords = ["io", "async", "non-blocking", "futures", "service"]
```
Comment (seanmonstar) [inline Cargo.toml]: I'd probably put `publish = false` on a newline, just to protect ourselves.

**Comment by jonhoo** [comment]
@seanmonstar oh, sorry, yeah, this is https://github.com/rust-lang/rust/issues/57524 which means cargo [refuses to compile everything](https://travis-ci.org/tower-rs/tower/jobs/481128112) now that 1.32 is stable.

File: Cargo.toml (hunk 6 above)
Comment (LucioFranco) [reply Cargo.toml]: done!
